File: python/S1_BASIC/L4_BASICMATHS/prime.py
class Solution:
    def __init__(self,num):
        self.num=num
    def isPrime(self):
        if self.num<0:
            return 
        if self.num==1 or self.num==0:
            print("no")
            return 
        flag=0
        for i in range(1,self.num//2+1):
            if self.num%i!=0:
                flag=1
        if flag==0:
            print("yes")
        else:
            print("no")
if __name__=="__main__":
    n=int(input("enter number :"))
    s=Solution(n)
    s.isPrime()

File: JAVA/S1_BASIC/L4_BASICMATHS/prime.java
class solution{
    int isPrime(int num){
        if(num<2)return 0;
        if(num==2)return 1;
        int flag=1;
        for(int i=2;i<=num/2;i++){
            if(num%i==0){
                flag=0;
            }
        }
        if(flag==0){
            return 0;
            

        }
        return 1;
    }
}
public class prime{
    public static void main(String[] args){
        solution s=new solution();
        int r=s.isPrime(29);
        if(r==1)System.out.print("yes");
        else{
            System.out.print("no");
        }
        
    }
}